File: solutions/javascript/pascals-triangle/1/pascals-triangle.js
export function rows(size) {
  if (size === 0) {
    return [];
  }

  const triangleRows = [[1]];
  if (size > 1) {
    const getNextRow = (row) => {
      const nextRow = [];
      nextRow.push(1);
      for (let i = 0; i <= row.length - 2; i++) {
        nextRow.push(row[i] + row[i + 1]);
      }
      nextRow.push(1);
      return nextRow;
    };
  
    while (size-- > 1) {
      triangleRows.push(getNextRow(triangleRows[triangleRows.length - 1]));
    }
  }

  return triangleRows;
};

File: solutions/dart/pascals-triangle/1/lib/pascals_triangle.dart
class PascalsTriangle {
  List<List<int>> rows(int rows) {
    List<int> nextRow = [];

    final triangle = List.generate(rows, (rowIndex) {
      nextRow = [
        1,
        if (nextRow.length > 1)
          for (int index = 1; index < nextRow.length; index++) nextRow[index - 1] + nextRow[index],
        if (rowIndex > 0) 1,
      ];

      return nextRow;
    });

    return triangle;
  }
}